SPECS/libcxx/libcxx.spec

```diff
@@ -0,0 +1,568 @@
+
+%package -n llvm-libunwind-static
+Summary: Static library for LLVM libunwind
+
+%description -n llvm-libunwind-static
+%{summary}.
+
+%prep
+%autosetup -p1 -n llvm-project-llvmorg-%{version}
+
+%build
+mkdir -p build
+cd build
+cmake \
+    -G Ninja \
+    -S ../runtimes \
+    -DLLVM_ENABLE_RUNTIMES="libcxx;libcxxabi;libunwind" \
+    -DCMAKE_INSTALL_PREFIX=%{_prefix} \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_MODULE_PATH="%{_libdir}/cmake/llvm;%{_datadir}/llvm/cmake/Modules" \
+    -DCMAKE_POSITION_INDEPENDENT_CODE=ON \
+    -DLIBCXX_STATICALLY_LINK_ABI_IN_STATIC_LIBRARY=ON \
+    -DLIBCXX_INCLUDE_BENCHMARKS=OFF \
+    -DLIBCXX_ENABLE_ABI_LINKER_SCRIPT=ON \
+    -DLIBUNWIND_INSTALL_INCLUDE_DIR=%{_includedir}/llvm-libunwind \
+    -DCXX_SUPPORTS_NOSTDLIBXX_FLAG=OFF
+
+%ninja_build cxx
+%ninja_build
+
+%install
+cd build
+%ninja_install
+
+# We can't install the unversionned path on default location because that would conflict with
+# https://src.fedoraproject.org/rpms/libunwind
+#
+# The versionned path has a different soname (libunwind.so.1 compared to
+# libunwind.so.8) so they can live together in %%{_libdir}
+#
+# ABI wise, even though llvm-libunwind's library is named libunwind, it doesn't
+# have the exact same ABI as gcc's libunwind (it actually provides a subset).
+rm %{buildroot}%{_libdir}/libunwind.so
+mkdir -p %{buildroot}/%{_libdir}/llvm-unwind/
+
+pushd %{buildroot}/%{_libdir}/llvm-unwind
+ln -s ../libunwind.so.1.0 libunwind.so
+popd
+
+%ldconfig_scriptlets
+
+%files
+%license libcxx/LICENSE.TXT
+%doc libcxx/CREDITS.TXT libcxx/TODO.TXT
+%{_libdir}/libc++.so.*
+
+%files devel
+%{_includedir}/c++/
+%exclude %{_includedir}/c++/v1/cxxabi.h
+%exclude %{_includedir}/c++/v1/__cxxabi_config.h
+%{_libdir}/libc++.so
+
+%files static
+%license libcxx/LICENSE.TXT
+%{_libdir}/libc++.a
+%{_libdir}/libc++experimental.a
+
+%files -n libcxxabi
+%license libcxxabi/LICENSE.TXT
+%doc libcxxabi/CREDITS.TXT
+%{_libdir}/libc++abi.so.*
+
+%files -n libcxxabi-devel
+%{_includedir}/c++/v1/cxxabi.h
+%{_includedir}/c++/v1/__cxxabi_config.h
+%{_libdir}/libc++abi.so
+
+%files -n libcxxabi-static
+%{_libdir}/libc++abi.a
+
+%files -n llvm-libunwind
+%license libunwind/LICENSE.TXT
+%{_libdir}/libunwind.so.1
+%{_libdir}/libunwind.so.1.0
+
+%files -n llvm-libunwind-devel
+%{_includedir}/llvm-libunwind/__libunwind_config.h
+%{_includedir}/llvm-libunwind/libunwind.h
+%{_includedir}/llvm-libunwind/libunwind.modulemap
+%{_includedir}/llvm-libunwind/mach-o/compact_unwind_encoding.h
+%{_includedir}/llvm-libunwind/mach-o/compact_unwind_encoding.modulemap
+%{_includedir}/llvm-libunwind/unwind.h
+%{_includedir}/llvm-libunwind/unwind_arm_ehabi.h
+%{_includedir}/llvm-libunwind/unwind_itanium.h
+%dir %{_libdir}/llvm-unwind
+%{_libdir}/llvm-unwind/libunwind.so
+
+%files -n llvm-libunwind-static
+%{_libdir}/libunwind.a
+
+%changelog
+* Wed Aug 07 2024 Andrew Phelps <anphel@microsoft.com> - 18.1.2-2
+- Initial Azure Linux import from Fedora 40 (license: MIT).
```

Comment:
It looks like we can get up to 18.1.8 version from [Fedora 40](https://packages.fedoraproject.org/pkgs/libcxx/libcxx/fedora-40-updates.html). Can we switch to that, so we get the latest possible version? Fedora 41 is even slightly further than that but only by `Release` number.

Reply:
I think that it would be best to keep all of these packages at the same version (libcxx, clang, llvm, lld, compiler-rt, lldb are all currently at 18.1.2). We can update them all at the same time in the future